### DISCUSSION TITLE
[ARM AArch64][Linux] How to use openssl to call linux crypto driver?

### DISCUSSION THREAD

**Discussion by thomas-coding** [openssl/openssl#22314] [Q&A] [answered]
Hi expert,

version: 1.1.1q

I tried to use the OpenSSL engine to call the Linux crypto driver, built with "enable-afalgeng", it can find the engine "afalg", but the Linux kernel sha256 algorithm driver is not called.

```
# openssl sha256 -engine afalg hello 
engine "afalg" set.
SHA256(hello)= e150a1ec81e8e93e1eae2c3a77e66ec6dbd6a3b460f89c1d08aecf422ee401a0
```

Thanks.

**Comment by t8m** [comment, marked as answer]
The afalg engine supports only AES-CBC ciphers, nothing else. It is rather just a proof-of-concept engine code not a full featured crypto engine.
